File: library/src/main/java/com/leocardz/link/preview/library/ImagePickingStrategy.java
package com.leocardz.link.preview.library;

import android.os.AsyncTask;

import org.jsoup.nodes.Document;

import java.util.HashMap;
import java.util.List;

/**
 * A strategy for how to select the images to return.
 */
public interface ImagePickingStrategy {

    void setImageQuantity(int imageQuantity);

    int getImageQuantity();

    List<String> getImages(AsyncTask asyncTask, Document doc, HashMap<String, String> metaTags);
}

File: linkpreview/src/main/java/com/leocardz/linkpreview/sample/library/ImagePickingStrategy.kt
package com.leocardz.linkpreview.sample.library

import org.jsoup.nodes.Document
import java.util.*

/**
 * A strategy for how to select the images to return.
 */
interface ImagePickingStrategy {

    var imageQuantity: Int

    fun getImages(getSource: GetSource, doc: Document, metaTags: HashMap<String, String>): List<String>
}
